src/extrovermectin.ts

```diff
@@ -334,22 +334,38 @@ const shortBanishes = [
 ];
 
 function banishBunny(): void {
-  const banishes = [
-    ...longBanishes,
-    ...(!have($item`miniature crystal ball`) ? shortBanishes : []),
-  ].filter((b) => b.available());
-
-  const banish = maxBy(banishes, (banish: Banish) => banish.price?.() ?? 0, true);
-  do {
-    banish.prepare?.();
-    garboAdventure(
-      $location`The Dire Warren`,
-      Macro.if_($monster`fluffy bunny`, banish.macro()).embezzler(),
+  if (getUsingFreeBunnyBanish()) {
+    do {
+      new Requirement(["100 Monster Level"], {
+        preventEquip: $items`carnivorous potted plant`,
+        forceEquip: $items`mafia middle finger ring`,
+      }).maximize();
+      garboAdventure(
+        $location`The Dire Warren`,
+        Macro.if_($monster`fluffy bunny`, Macro.skill($skill`Show them your ring`)).embezzler(),
+      );
+    } while (
+      "fluffy bunny" !== get("lastEncounter") &&
+      !get("banishedMonsters").includes("fluffy bunny")
```

Comment:
I'm not sure how I feel about separating this from the main body of the code instead of formatting the free banish like the other banishes and forcing it to be chosen if available (which will already happen, because it's free)